file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/service/impl/ItemActionEventServiceImpl.java

```diff
@@ -86,8 +86,16 @@ public void directoryMoved(MoveDirectoryAction action) {
 	@Transactional(readOnly = false)
 	public void fileUploaded(UploadFileAction action) {
 		registerTransactionSynchronization(action, UPLOAD_FILE);
-		FileItem fileItem = itemsActionBusiness.saveFileItem(action);
-		itemsActionBusiness.saveFileItemAction(fileItem.getId(), UPLOAD_FILE);
+		File file = new File(action.getPath(), action.getItemName());
```

Comment:
There are a number of issues here:
1.  The service only handles actions already performed on the disk; when the fileUploaded method is called, the file is already uploaded on the disk.
2. We should never perform I/O operations inside a transactional layer. If the transaction fails, we rollback everything, but the stream of data could be already written somewhere (this applies to not just file stream but any type of streams)
3. Logic about file exists should be moved in the handler (don't remember exactly the name) that handles upload of file on the disk.
4. Use Java NIO2 only :)